### ISSUE TITLE
first tick label on categorical x-axis misplaced when labels are rotated

### ISSUE THREAD

**Issue by JanHomann** [holoviz/hvplot#925]
### Versions
```
hvplot      0.8.1
holoviews   1.15.1
pandas      1.5.0
python      3.9.13
jupyterlab  3.4.5

Chrome Browser
```

### Description
The first tick label on a categorical x-axis is not correctly placed along the horizontal axis when slanted (so not 0 or 90 degrees). It is much closer to the second tick label than it should be. This becomes especially obvious when working with a big set of labels or alternatively when the plot is made very narrow. I have tested `bar`, `area`, `line`, and  `scatter`.

### Example
```python
import hvplot.pandas
import pandas as pd


df1 = pd.DataFrame([["Steven Spielberg",    5],
                    ["Alfred Hitchcock",    4],
                    ["Stanley Kubrick",     4],
                    ["Billy Wilder",        4],
                    ["John Huston",         3]],
            columns=['Director','Frequency'])

df1.hvplot.bar(rot=45, x='Director', y='Frequency', width=150)
```

### Stack traceback
No error message.

### Screenshots
<img width="503" alt="Screen Shot 2022-10-09 at 1 21 28 AM" src="https://user-images.githubusercontent.com/35338267/194739468-b0563003-daeb-45b3-bfbf-382adf1892eb.png">

**Comment by jbednar** [comment]
Thanks. My guess is that this is a Bokeh issue rather than HoloViews, since I don't think HoloViews messes with the spacing of individual labels. Making a pure-Bokeh example might be difficult, but maybe this is enough for you to go on, @mattpap?

**Comment by mattpap** [comment]
Looks similar to this https://github.com/bokeh/bokeh/issues/12069 one, which is easily reproducible in bokeh. I didn't have chance yet to investigate why this happens.

**Comment by JanHomann** [comment]
> Looks similar to this [bokeh/bokeh#12069](https://github.com/bokeh/bokeh/issues/12069) one, which is easily reproducible in bokeh. I didn't have chance yet to investigate why this happens.

I think that the plotting library sees the vertical y-axis label as the outer left edge of the plot, so any tick label that extends beyond that gets shifted inwards. But I could be wrong.

**Comment by jbednar** [comment]
The Bokeh issue suggests it's fixed in bokeh 3.0, which is imminent, though it may be some time before holoviz tools support the big changes that 3.0 brings.

**Comment by mattpap** [comment]
No, it's a secondary issue that was fixed (no more "invalid bbox" messages). The misplaced tick label is still a problem.

**Comment by jbednar** [comment]
Ah, the ever-ambiguous "this" was fixed. :-/
